### ISSUE TITLE
[CLI]: Python agent missing traceback

### ISSUE THREAD

**Issue by collinmccarthy** [wandb/client#3576]
### Describe the bug

I'm having the same issue as in #2387, specifically for the python (not CLI) version of Wandb agent when --count=1. 

I'm using the VSCode debugger, and when an exception in my function is raised I only see something like the following (and nothing in the two log files relating to my own source code).

```
wandb: Find logs at: /tmp/wandb/wandb/run-20220425_212502-6q4aki1t/logs
wandb: ERROR Run 6q4aki1t errored: ValueError('too many values to unpack (expected 2)')
```

Stepping through the [python code](https://github.com/wandb/client/blob/272ce0aed8610df9e012b4645d00bac96d90d188/wandb/agents/pyagent.py#L228), what gets printed out is `repr(exc)` not the traceback, and the exception is never re-raised. None of the other if/else blocks below this that use `self._exceptions` get entered and the code just hits a [return statement](https://github.com/wandb/client/blob/272ce0aed8610df9e012b4645d00bac96d90d188/wandb/agents/pyagent.py#L261) and exits. 

Is this a bug or are you intentionally trying not to log a traceback? It makes development almost impossible when you can't see the traceback without setting a breakpoint in the Wandb codebase.

Thanks,
Collin

### Environment

WandB version: 0.12.15

OS: Linux

Python version: 3.8.13

Versions of relevant libraries:

**Comment by collinmccarthy** [comment]
Specifically, for my use case it would be sufficient to modify the error logging [here](https://github.com/wandb/client/blob/272ce0aed8610df9e012b4645d00bac96d90d188/wandb/agents/pyagent.py#L228) to be something like:

```python
logger.error(f"Run {run_id} errored: {repr(exc)}\n{traceback.print_exc()}")
wandb.termerror(f"Run {run_id} errored: {repr(exc)}\n{traceback.print_exc()}")
```

instead of

```python
logger.error(f"Run {run_id} errored: {repr(exc)}")
wandb.termerror(f"Run {run_id} errored: {repr(exc)}")
```

**Comment by collinmccarthy** [comment]
One more related question, why are we forcing the Python agent to be threaded? Specifically why use `pyagent` [here](https://github.com/wandb/client/blob/272ce0aed8610df9e012b4645d00bac96d90d188/wandb/wandb_agent.py#L575) rather than let it use the normal multiprocessing agent by calling `run_agent()` two lines below?

**Comment by collinmccarthy** [comment]
Hi Leslie, I was just trying to get my workflow functioning, I don't have time to wait for a bug fix with some deadlines approaching. I've since stopped using the agent as I can't get consistent behavior out of it (repeated runs, multiprocessing issues, etc.). I opened another issue but decided to move on from the agent for now. Thanks.

Closing since you've opened an internal ticket for the traceback.
